### DISCUSSION TITLE
example FromRequest process for User Authentication. querying database each process

### DISCUSSION THREAD

**Discussion by eisenhaus335** [rwf2/Rocket#1764] [Questions]
I trying to figure out on how to make FromRequest guard while also using database service. I use some idea where <https://api.rocket.rs/v0.5-rc/rocket/request/trait.FromRequest.html> are explained here on how to handle user guard. I cant figure out how to make this making 'guard' from the examples/database, there are no example on processing database inside FromRequest . rust-analyzer also not showing good information due macro is complex there, or perhaps is just me that doesn't understand it

Here is the link for my recent work <https://github.com/eisenhaus335/rockety_backend>

**Comment by jebrosen** [comment]
>  I cant figure out how to make this making 'guard' from the examples/database, there are no example on processing database inside FromRequest.

The page you linked to does access the database inside of `FromRequest`, in particular this line:

```rust
let db = try_outcome!(request.guard::<Database>().await);
```

The [`guard()`](https://api.rocket.rs/v0.5-rc/rocket/request/struct.Request.html#method.guard) function lets you run any request guard; when you call `guard().await`, you can use the database handle the same way as you would have in a route.

Can you be more specific about which line(s) of code you are having trouble with?
